api/v1beta1/federation_webhook.go

```diff
@@ -108,6 +110,41 @@ func (r *Federation) ValidateDelete(ctx context.Context, client client.Client, u
 	return nil
 }
 
+// validateMembers If a member is deleted, he is the initiator of the network or a member of the channel,
+// the operation is not allowed
+func validateMembers(ctx context.Context, c client.Client, oldMembers, newMembers []Member) error {
+	deleted := make(map[string]struct{})
+	for _, member := range newMembers {
+		deleted[member.Name] = struct{}{}
+	}
+	chList := &ChannelList{}
+	if err := c.List(ctx, chList); err != nil && !errors2.IsNotFound(err) {
+		return err
+	}
+	networkList := &NetworkList{}
+	if err := c.List(ctx, networkList); err != nil && !errors2.IsNotFound(err) {
+		return err
+	}
+
+	for _, member := range oldMembers {
+		if _, ok := deleted[member.Name]; !ok {
+			for _, i := range networkList.Items {
+				if i.Labels != nil && i.Labels[NETWORK_INITIATOR_LABEL] == member.Name {
```

Comment:
maybe it's better to use network & channel selector instead of listing all networks/channels.
The network/channel list can be very long...